app/controllers/api/v1/documents_controller.rb

```diff
@@ -24,25 +24,27 @@ def destroy
 
     private
 
-      PERMITTED_DOCUMENT_PARAMS = [
-        attributes: %i[file],
-      ].freeze
-
-      def document_params
-        params.require(:data).permit(PERMITTED_DOCUMENT_PARAMS).to_h
+      def document_attributes
+        document_params[:attributes].merge(documentable: profile)
       end
 
-      def document_attributes
-        document_params[:attributes].merge(
-          move: Move.find_by(id: params.dig(:move_id)),
-        )
+      def document_params
+        params.require(:data).permit(attributes: %i[file]).to_h
       end
 
     protected
 
       def set_restricted_request_content_type
         @restricted_request_content_type = 'multipart/form-data'
       end
+
+      def profile
+        move_id = params[:move_id]
+
```

Comment:
this is just a matter of style, you could use a guard clause
`return unless move_id.present?`

Reply:
I know about that style. I did find this more readable. If this was a more complex method I'd probably go for an early return.